packages/twenty-front/src/modules/settings/security/hooks/useCreateSSOIdentityProvider.ts

```diff
@@ -0,0 +1,61 @@
+import { useSetRecoilState } from 'recoil';
+import {
+  CreateOidcIdentityProviderMutationVariables,
+  CreateSamlIdentityProviderMutationVariables,
+  IdpType,
+  useCreateOidcIdentityProviderMutation,
+  useCreateSamlIdentityProviderMutation,
+} from '~/generated/graphql';
+import { SSOIdentitiesProvidersState } from '@/settings/security/states/SSOIdentitiesProviders.state';
+
+export const useCreateSSOIdentityProvider = () => {
+  const [createOidcIdentityProviderMutation] =
+    useCreateOidcIdentityProviderMutation();
+  const [createSamlIdentityProviderMutation] =
+    useCreateSamlIdentityProviderMutation();
+
+  const setSSOIdentitiesProviders = useSetRecoilState(
+    SSOIdentitiesProvidersState,
+  );
+
+  const createSSOIdentityProvider = async <T extends IdpType>({
+    type,
+    input,
+  }: T extends 'OIDC'
+    ? {
+        type: 'OIDC';
+        input: CreateOidcIdentityProviderMutationVariables['input'];
+      }
+    : {
+        type: 'SAML';
+        input: CreateSamlIdentityProviderMutationVariables['input'];
+      }) => {
+    if (type === 'OIDC') {
+      return await createOidcIdentityProviderMutation({
+        variables: { input },
+        onCompleted: (data) => {
+          setSSOIdentitiesProviders((existingProvider) => [
```

Comment:
@charlesBochet do you have an idea why this call does not refresh the list of providers used in the component `SettingsSSOIdentitiesProvidersListCard.tsx`

Reply:
not sure, let's investigate together tomorrow